akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -268,6 +268,48 @@ private[akka] final case class Grouped[T](n: Int) extends PushPullStage[T, immut
 /**
  * INTERNAL API
  */
+private[akka] final case class Sliding[T](n: Int, step: Int) extends PushPullStage[T, immutable.Seq[T]] {
+  private val buf = {
+    val b = Vector.newBuilder[T]
+    b.sizeHint(n)
+    b
+  }
+
+  override def onPush(elem: T, ctx: Context[immutable.Seq[T]]): SyncDirective = {
+    buf += elem
+    val emit = buf.result()
+    if (emit.size < n) {
+      ctx.pull()
+    } else if (emit.size == n) {
+      ctx.push(emit)
```

Comment:
`buf.clear` would delete elements which are needed in a possible next step of the window. I don't think its possible to do it as in `Grouped`, right?